diff --git a/TirelessTracker/app/Events.hs b/TirelessTracker/app/Events.hs
index b1e8910..93282e4 100644
--- a/TirelessTracker/app/Events.hs
+++ b/TirelessTracker/app/Events.hs
@@ -62,13 +62,16 @@ eventAction command (ESave empty) state =
             return [ESave fileName]
 
 eventAction command EStats (AppState matches) =
-        let 
-            (w, l, d) = winLossDrawPerc matches
-            numMatches = length matches
-        in do
-            putStrLn $ show w ++ "% won, " ++ show l ++ "% lost, " ++ show d ++ "%. " ++ show numMatches ++ " total games.\n"
-            putStrLn $ oppDeckWinPercReport matches ++"\n"
-            return [EStats]
+    let 
+        (w, l, d) = winLossDrawPerc matches
+        numMatches = length matches
+    in do
+        putStrLn $ show w ++ "% won, " ++ show l ++ "% lost, " ++ show d ++ "%. " ++ show numMatches ++ " total games.\n"
+        putStrLn $ oppDeckWinPercReport matches ++"\n"
+        return [EStats]
+
+--eventAction command ESort (AppState matches) =
+    
 
 eventAction command EHelp state = do
     helpMessage
@@ -101,25 +104,28 @@ parseCommandsRes (c:commands) =
 
 commandFromString :: String -> String -> ([Match] -> [Match])
 commandFromString command args =
-    \matches -> filter (commandSpecifiers command args) matches
+    let
+        spec = commandSpecifiers command args
+    in
+    \matches -> filter (\m -> (spec m) == EQ) matches
 
-commandSpecifiers :: String -> String -> (Match -> Bool)
+commandSpecifiers :: String -> String -> (Match -> Ordering)
 commandSpecifiers command args =
     if command == "-d" then
         let
             deck = Match {myDeck = args, oppDeck = "", result = (Win, (0, 0)),date = makeDay 0 0 0,eventType = ""}
         in
-            (\m -> compareMatch [(\m -> D $ myDeck m )] deck m == EQ)
+            (\m -> compareMatch [(\m -> D $ myDeck m )] deck m)
     else if command == "-o" then
         let
             deck = Match {myDeck = "", oppDeck = args, result = (Win, (0, 0)),date = makeDay 0 0 0,eventType = ""}
         in
-            (\m -> compareMatch [(\m -> D $ oppDeck m )] deck m == EQ)
+            (\m -> compareMatch [(\m -> D $ oppDeck m )] deck m)
     -- else if command == "-w" then
     --     let
     --         deck = Match {myDeck = "", oppDeck = "", result = (Win, (0, 0)),date = makeDay 0 0 0,eventType = ""}
     --     in
-    --         (\m -> compareMatch [(\m -> D $ myDeck m )] deck m == EQ)
+    --         (\m -> compareMatch [(\m -> D $ myDeck m )] deck m)
     else if command == "-t" then
         if (length $ words args) == 3 then
             let
@@ -129,13 +135,13 @@ commandSpecifiers command args =
                 day = fromIntegral $ argList !! 2
                 deck = Match {myDeck = args, oppDeck = "", result = (Win, (0, 0)),date = makeDay year month day,eventType = ""}
             in
-                (\m -> compareMatch [(\m -> Dy $ date m)] deck m == EQ)
+                (\m -> compareMatch [(\m -> Dy $ date m)] deck m)
         else
-            \m -> True
+            \m -> EQ
     else if command == "-e" then
         let
             deck = Match {myDeck = args, oppDeck = "", result = (Win, (0, 0)),date = makeDay 0 0 0,eventType = args}
         in
-            (\m -> compareMatch [(\m -> D $ eventType m )] deck m == EQ)
+            (\m -> compareMatch [(\m -> D $ eventType m )] deck m)
     else
-        \m -> True
+        \m -> EQ
diff --git a/TirelessTracker/res/HelpMessage.in b/TirelessTracker/res/HelpMessage.in
index 5e21515..9e51f36 100644
--- a/TirelessTracker/res/HelpMessage.in
+++ b/TirelessTracker/res/HelpMessage.in
@@ -5,11 +5,17 @@ Commands:
     
 help                       - Displays help information
 add [match]                - Adds [match] if it is in MATCHFORMAT
-show                       - Displays the list of matches
+show [(optional) command]  - Displays the list of matches
     -d [deck name] - Only show matches played with [deck name]
     -o [deck name] - Only show matches played against [deck name]
     -t [year] [month] [day] - Only show matches played on the date
     -e [event name] - only show matches played in [event name]
+stats                       - Show win loss and draw statistics for the total and all matches-ups 
+sort                        - Sort the matches based on the command specifier
+    -d - Sort matches by deck played
+    -o - Sort matches by opponent deck 
+    -t - Sort matches by date
+    -e - Sort matches by event name
 load [file name]            - Load a file from [file name] location
 save [(optional)file name]  - saves matches at [file name] if specified, otherwise to the default location
 exit                       - Exits Tireless Tracker
